internal/depgraph/mod.go

```diff
@@ -0,0 +1,324 @@
+
+Examples:
+
+./depgrah --modname "go.dedis.ch/dela" -o graph.dot -F ./
+./depgrah --config internal/depgraph/dep.yml -o graph.dot -F ./
+
+The following commands can be used to generate a visual representation from the
+output of depgraph using DOT:
+
+dot -Tpdf graph.dot -o graph.pdf
+dot -Gdpi=300 -Tpng graph.dot -o graph.png -Gsplines=ortho`,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "config",
+				Aliases: []string{"c"},
+				Usage:   "the path to a yaml config file",
+			},
+			&cli.StringFlag{
+				Name: "modname",
+				Usage: "the module name, convenient if one doesn't want to " +
+					"provide a config file. Overwrites value from the config " +
+					"file if provided.",
+			},
+			&cli.StringFlag{
+				Name:    "out",
+				Aliases: []string{"o"},
+				Usage:   "if provided will save the result to the specified file",
+			},
+			&cli.BoolFlag{
+				Name:    "force",
+				Aliases: []string{"F"},
+				Usage:   "overwrites the output file",
+			},
+			&cli.BoolFlag{
+				Name:    "withTest",
+				Aliases: []string{"t"},
+				Usage:   "includes the test files",
+			},
+		},
+		Action: run,
+	}
+
+	err := app.Run(os.Args)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// run is the main action of the CLI.
+func run(c *cli.Context) error {
+
+	searchDir := c.Args().First()
+	if searchDir == "" {
+		return xerrors.Errorf("please provide the folder path")
+	}
+
+	config := config{}
+
+	configPath := c.String("config")
+	if configPath != "" {
+		configBuf, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return xerrors.Errorf("failed to read config file: %v", err)
+		}
+
+		err = yaml.Unmarshal([]byte(configBuf), &config)
+		if err != nil {
+			return xerrors.Errorf("failed to unmarshal config: %v", err)
+		}
+
+		config.Modname = config.Modname + "/"
+	}
+
+	if c.String("modname") != "" {
+		// we add a "/" to build the full package name. If the module name is
+		// mod.ch/module, then a package 'pancake' inside it should be
+		// mod.ch/module/pancake, but the parsing will only extract 'pancake'.
+		config.Modname = c.String("modname") + "/"
+	}
+
+	fset := token.NewFileSet()
+	out := os.Stdout
+
+	if c.String("out") != "" {
+
+		_, err := os.Stat(c.String("out"))
+		if !os.IsNotExist(err) && !c.Bool("force") {
+			return xerrors.Errorf("file '%s' already exist, use '-F' to "+
+				"overwrite", c.String("out"))
+		}
+
+		out, err = os.Create(c.String("out"))
+		if err != nil {
+			return xerrors.Errorf("failed to create output file: %v", err)
+		}
+	}
+
+	// We build a bag of interfaces with a map.
+	interfaces := make(map[string]struct{})
+	for _, it := range config.Interfaces {
+		interfaces[it] = struct{}{}
+	}
+
+	// links will contain, for every package, a bag of dependencies. The bag is
+	// done with a dummy map.
+	links := make(map[string]map[string]struct{})
+
+	// parseFile will be called recursively on each file and folder
+	parseFile := func(path string, f os.FileInfo, err error) error {
```

Comment:
same reasoning, to improve readability: 
- I would extract this as a separate function, taking config & links as an extra parameter
- call filepath.Walk with a simple function wrapper which inserts the parameters